zio-schema-json/shared/src/test/scala-2/zio/schema/codec/JsonCodecSpec.scala

```diff
@@ -140,6 +141,15 @@ object JsonCodecSpec extends ZIOSpecDefault {
           Enumeration(StringValue("foo")),
           JsonCodec.Encoder.charSequenceToByteChunk("""{"oneOf":{"StringValue":{"value":"foo"}}}""")
         )
+      },
+      test("ADT with annotation") {
```

Comment:
We need a test for decoding, too! To ensure we can round-trip: go from our case class to JSON, and go back from our JSON to the same case class.